examples/gpt-2/main.cpp

```diff
@@ -360,11 +403,19 @@ bool gpt2_model_load(const std::string & fname, gpt2_model & model, gpt_vocab &
                 return false;
             }
 
-            fin.read(reinterpret_cast<char *>(tensor->data), ggml_nbytes(tensor));
+            // read into a temporary buffer first, then copy to the tensor
+            // TODO: read directly into the tensor if the backend is CPU
```

Comment:
Isn't this a simple if ? :)  And is there a simple way to check whether a backend is cpu or gpu ones it is created

Reply:
The problem is that there isn't an easy way to tell if  backend is the CPU backend. The only way to do this at the moment is by checking the name of the backend (as is done in `gpt2_eval` to set the number of threads), but I don't really like that. I am thinking about adding a backend id enum, but that doesn't seem great either because it will need to be modified every time a new backend is added.